### PR TITLE
json and external templates

### PR DESCRIPTION
**Pull request by tom--**
One change dumps the info from docblocks to JSON.

The other allows templates to be outside the extension's root.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
+.idea
```
Comment (cebe) [inline .gitignore]: this should not be here but in your global gitignore.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
...
 /vendor
-composer.lock
+/output
```
Comment (cebe) [inline .gitignore]: composer.lock should not be part of the repo so it should be ignored.
output is also not really needed as it is not a default output directory.

File: templates/json/ApiRenderer.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+use yii\apidoc\models\Context;
+use yii\apidoc\renderers\ApiRenderer as BaseApiRenderer;
+use yii\base\ViewContextInterface;
+use Yii;
+
+/**
+ * The class for outputting documentation data structures as a JSON text.
+ *
+ * @author Carsten Brandt <mail@cebe.cc>
```
Comment (cebe) [inline templates/json/ApiRenderer.php]: that's not true :)

File: .gitignore (hunk 1 above)
Comment (samdark) [inline .gitignore]: Well, we have it in framework https://github.com/yiisoft/yii2/blob/master/.gitignore#L2

Comment (tom--) [inline .gitignore]: Yii Framework is a wonderful thing but not every single aspect of it should be understood as a guide to best practices :P

Comment (samdark) [inline .gitignore]: Well, yeah :)

**Comment by cebe** [comment]
Thank you!

**Pull request closed without merging**